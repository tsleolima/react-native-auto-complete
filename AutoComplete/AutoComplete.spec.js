/* global describe, it, expect, beforeEach, jest */

import React from 'react'
import { shallow } from 'enzyme'

import AutoComplete from '../'

describe('AutoComplete', () => {
  let fakeSuggestions
  beforeEach(() => {
    fakeSuggestions = [{
      text: '0.4'
    }, {
      text: '0.5'
    }, {
      text: '0.6'
    }]
  })

  it('renders correctly', () => {
    expect(shallow(
      <AutoComplete />
    )).toMatchSnapshot()
  })

  it('only renders the suggestions that have a similarity score greater than or equal to the minimum score passed', () => {
    const params = {
      value: '0.5',
      suggestions: fakeSuggestions,
      suggestionObjectTextProperty: 'text',
      minimumSimilarityScore: 0.5
    }

    const wrapper = shallow(
      <AutoComplete {...params} />
    )

    wrapper.instance().componentDidMount()
    wrapper.instance().forceUpdate()

    expect(wrapper.find('TouchableHighlight').length).toEqual(2)

    expect(wrapper.find({suggestionText: '0.5'}).length).toEqual(1)
    expect(wrapper.find({suggestionText: '0.6'}).length).toEqual(1)
  })

  it('correctly changes the rendered suggestions when receiving new props', () => {
    let params = {
      value: '0.5',
      suggestions: fakeSuggestions,
      suggestionObjectTextProperty: 'text',
      minimumSimilarityScore: 0.5
    }

    const wrapper = shallow(
      <AutoComplete {...params} />
    )

    wrapper.instance().componentDidMount()
    wrapper.instance().forceUpdate()

    expect(wrapper.find('TouchableHighlight').length).toEqual(2)

    params.value = '0.6'
    wrapper.setProps(params)

    expect(wrapper.find('TouchableHighlight').length).toEqual(1)
    expect(wrapper.find({suggestionText: '0.6'}).length).toEqual(1)
  })

  it('hides the suggestions and calls the select callback when a suggestion is selected', () => {
    const params = {
      value: '0.5',
      suggestions: fakeSuggestions,
      suggestionObjectTextProperty: 'text',
      onSelect: jest.fn(),
      minimumSimilarityScore: 0.5
    }

    const wrapper = shallow(
      <AutoComplete {...params} />
    )

    wrapper.instance().componentDidMount()
    wrapper.instance().forceUpdate()

    expect(wrapper.find('TouchableHighlight').length).toEqual(2)

    wrapper.find({suggestionText: '0.5'}).simulate('press')

    expect(params.onSelect.mock.calls.length).toEqual(1)
    expect(params.onSelect.mock.calls[0]).toEqual([{
      text: '0.5'
    }])
    wrapper.instance().forceUpdate()
    expect(wrapper.find('TouchableHighlight').length).toEqual(0)
  })

  it('correctly calls the onChangeText callback when the text changes', () => {
    const params = {
      onChangeText: jest.fn()
    }

    const wrapper = shallow(
      <AutoComplete {...params} />
    )

    wrapper.find('TextInput').simulate('changeText', 'Some Text')

    expect(params.onChangeText.mock.calls.length).toEqual(1)
    expect(params.onChangeText.mock.calls[0]).toEqual(['Some Text'])
  })
})
