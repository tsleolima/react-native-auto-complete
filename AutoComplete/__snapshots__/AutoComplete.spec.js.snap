exports[`AutoComplete renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "flex": 1,
          }
    }>
    <TextInput
        comparationFuzziness={0.5}
        minimumSimilarityScore={0.6}
        onChangeText={[Function]}
        style={
            Object {
                "fontSize": 15,
              }
        } />
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "flex": 1,
                }
      }>
      <TextInput
            comparationFuzziness={0.5}
            minimumSimilarityScore={0.6}
            onChangeText={[Function]}
            style={
                  Object {
                        "fontSize": 15,
                      }
            } />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AutoComplete
        comparationFuzziness={0.5}
        minimumSimilarityScore={0.6} />,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AutoComplete {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "filterSugestions": [Function],
        "getSuggestionText": [Function],
        "isSimilar": [Function],
        "onChangeText": [Function],
        "props": Object {
          "comparationFuzziness": 0.5,
          "minimumSimilarityScore": 0.6,
        },
        "refs": Object {},
        "renderSuggestions": [Function],
        "shouldFilterSuggestions": [Function],
        "state": null,
        "suggestionClick": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }>
          <TextInput
                    comparationFuzziness={0.5}
                    minimumSimilarityScore={0.6}
                    onChangeText={[Function]}
                    style={
                              Object {
                                        "fontSize": 15,
                                      }
                    } />
</View>,
        "_renderedOutput": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }>
          <TextInput
                    comparationFuzziness={0.5}
                    minimumSimilarityScore={0.6}
                    onChangeText={[Function]}
                    style={
                              Object {
                                        "fontSize": 15,
                                      }
                    } />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AutoComplete
    comparationFuzziness={0.5}
    minimumSimilarityScore={0.6} />,
}
`;
